extendr-api/src/graphics/device_driver.rs

```diff
@@ -265,6 +270,15 @@ pub trait DeviceDriver: std::marker::Sized {
         0
     }
 
+    /// A callback function that returns the coords of the event
+    fn locator(&mut self, x: &mut f64, &mut y: f64, dd: DevDesc) {}
+
+    /// A callback function for X11_eventHelper.
+    /// Argument `code` should, ideally, be of type c_int,
+    /// but compiler throws erors. It should be ok to use
+    /// i32 here.
```

Comment:
Curious.

Reply:
Sorry for keeping this unfixed for that long.
I will try to get back to this in the upcoming week -- had some issues with running github actions locally via act.

Reply:
No worries! Please feel free to ask for any help.

Reply:
Done.
The commit history looks a bit awkward though.

Reply:
I have the following error, which I cannot resolve.
I'm not even sure why `u32` is expected in the first place.

*Output*:
> Error[E0308]: mismatched types
>    --> /home/stlk/projects/oss/extendr/extendr-api/src/graphics/device_driver.rs:794:42
>     |
> 794 |             (*p_dev_desc).locator = Some(device_driver_locator::<T>);
>     |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `()`
>     |
>     = note: expected fn pointer `unsafe extern "C" fn(_, _, _) -> u32`
>                   found fn item `unsafe extern "C" fn(_, _, _) {device_driver_locator::<T>}`
> 
> For more information about this error, try `rustc --explain E0308`.